src/packeteer.rs

```diff
@@ -60,4 +63,36 @@ impl<'a> Packeteer<'a> {
 
         Ok(packet)
     }
+
+    /// Block until we receive the next unique [`GameTickPacket`],
+    /// and then return it.
+    ///
+    /// # Errors
+    ///
+    /// This function returns an error if ten seconds pass without a new
+    /// packet being received. The assumption is that the game froze or
+    /// crashed, and waiting longer will not help.
+    pub fn next_flatbuffer(&mut self) -> Result<GameTickPacket, Box<Error>> {
+        let started = Instant::now();
+
+        loop {
+            self.ratelimiter.wait();
+
+            if let Some(packet) = self.rlbot.update_live_data_packet_flatbuffer() {
+                // Wait until another "tick" has happened so we don't return duplicate data.
+                let game_time = packet
+                    .gameInfo()
+                    .ok_or("Missing gameInfo")?
+                    .secondsElapsed();
+                if game_time != self.prev_game_time {
+                    self.prev_game_time = game_time;
+                    return Ok(packet);
+                }
+            }
+
+            if Instant::now() - started > Duration::from_secs(10) {
+                return Err(From::from("no packet received after ten seconds"));
```

Comment:
10 seconds because I noticed I was hitting this error after doing an injection earlier and restarting the bot.

Reply:
Yeah I don't know this is the right place for this myself, but I don't have an alternative in mind just yet.